Crash when opening the settings

### Minecraft Version

1.21.0

### Skyblocker Version

skyblocker-1.21.1-beta-da041ea+1.21 

### Reproduction Steps

Run the game, open settings, and it crashes.

### Crash Report file and latest.log

```shell
[11:04:04] [Render thread/INFO]:
 Generating YACL screen
[11:04:04] [Render thread/ERROR]:
 Reported exception thrown!
	at net.minecraft.class_312.method_22686(class_312.java:196) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:108) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22684(class_312.java:196) ~[client-intermediary.jar:?]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.3.jar:?]
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3509) ~[lwjgl-glfw-3.3.3.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:184) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1310) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:882) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:256) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.0.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.0.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.16.0.jar:?]
Caused by: java.lang.NullPointerException: Cannot invoke "String.isEmpty()" because "this.inputField" is null
	at dev.isxander.yacl3.gui.controllers.string.StringControllerElement.getValueText(StringControllerElement.java:461) ~[YetAnotherConfigLib-3.5.0+1.21-fabric.jar:?]
	at dev.isxander.yacl3.gui.controllers.string.StringControllerElement.setDimension(StringControllerElement.java:436) ~[YetAnotherConfigLib-3.5.0+1.21-fabric.jar:?]
	at dev.isxander.yacl3.gui.controllers.string.StringControllerElement.<init>(StringControllerElement.java:45) ~[YetAnotherConfigLib-3.5.0+1.21-fabric.jar:?]
	at dev.isxander.yacl3.gui.controllers.string.IStringController.provideWidget(IStringController.java:42) ~[YetAnotherConfigLib-3.5.0+1.21-fabric.jar:?]
	at dev.isxander.yacl3.gui.OptionListWidget.refreshOptions(OptionListWidget.java:75) ~[YetAnotherConfigLib-3.5.0+1.21-fabric.jar:?]
	at dev.isxander.yacl3.gui.OptionListWidget.<init>(OptionListWidget.java:40) ~[YetAnotherConfigLib-3.5.0+1.21-fabric.jar:?]
	at dev.isxander.yacl3.gui.YACLScreen$CategoryTab.<init>(YACLScreen.java:359) ~[YetAnotherConfigLib-3.5.0+1.21-fabric.jar:?]
	at dev.isxander.yacl3.gui.YACLScreen.lambda$init$4(YACLScreen.java:89) ~[YetAnotherConfigLib-3.5.0+1.21-fabric.jar:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.toArray(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.toArray(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.toList(Unknown Source) ~[?:?]
	at dev.isxander.yacl3.gui.YACLScreen.method_25426(YACLScreen.java:90) ~[YetAnotherConfigLib-3.5.0+1.21-fabric.jar:?]
	at net.minecraft.class_437.method_25423(class_437.java:309) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1507(class_310.java:1143) ~[client-intermediary.jar:?]
	at com.terraformersmc.modmenu.gui.widget.entries.ModListEntry.openConfig(ModListEntry.java:185) ~[modmenu-11.0.1.jar:?]
	at com.terraformersmc.modmenu.gui.widget.entries.ModListEntry.method_25402(ModListEntry.java:175) ~[modmenu-11.0.1.jar:?]
	at com.terraformersmc.modmenu.gui.widget.ModListWidget.method_25402(ModListWidget.java:298) ~[modmenu-11.0.1.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:38) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:107) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:431) ~[client-intermediary.jar:?]
	... 14 more
```


### Additional context

_No response_

Minecraft Version
1.21.0

Skyblocker Version
skyblocker-1.21.1-beta-da041ea+1.21

slightly different crash report

[22:14:04] [Render thread/INFO]:
 Generating YACL screen
[22:14:04] [Render thread/ERROR]:
 Reported exception thrown!
net.minecraft.class_148: mouseClicked event handler
	at net.minecraft.class_437.method_25412(class_437.java:436) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1601(class_312.java:107) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22686(class_312.java:196) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:108) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22684(class_312.java:196) ~[client-intermediary.jar:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.3.jar:?]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.3.jar:?]
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3438) ~[lwjgl-glfw-3.3.3.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.pollEvents(RenderSystem.java:150) ~[client-intermediary.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:168) ~[client-intermediary.jar:?]
	at net.minecraft.class_1041.method_15998(class_1041.java:303) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1307) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:882) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:256) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.0.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.0.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.16.0.jar:?]
Caused by: java.lang.NullPointerException: Cannot invoke "String.isEmpty()" because "this.inputField" is null
	at dev.isxander.yacl3.gui.controllers.string.StringControllerElement.getValueText(StringControllerElement.java:461) ~[yet_another_config_lib_v3-3.5.0+1.21-fabric-bbf2959bd3405562.jar:?]

[skyblocker.json](https://github.com/user-attachments/files/16488853/skyblocker.json)

--- a/src/main/java/de/hysky/skyblocker/SkyblockerMod.java
+++ b/src/main/java/de/hysky/skyblocker/SkyblockerMod.java
@@ -2,6 +2,8 @@ package de.hysky.skyblocker;
 
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
+
+import de.hysky.skyblocker.config.ConfigNullFieldsFix;
 import de.hysky.skyblocker.config.SkyblockerConfigManager;
 import de.hysky.skyblocker.config.datafixer.ConfigDataFixer;
 import de.hysky.skyblocker.debug.Debug;
@@ -109,6 +111,7 @@ public class SkyblockerMod implements ClientModInitializer {
         ConfigDataFixer.apply();
         Utils.init();
         SkyblockerConfigManager.init();
+        ConfigNullFieldsFix.init(); //DO NOT INIT ANY CLASS THAT USES CONFIG FIELDS BEFORE THIS!
         SkyblockerScreen.initClass();
         ProfileViewerScreen.initClass();
         Tips.init();

--- a/src/main/java/de/hysky/skyblocker/config/ConfigNullFieldsFix.java
+++ b/src/main/java/de/hysky/skyblocker/config/ConfigNullFieldsFix.java
@@ -43,10 +43,8 @@ public class ConfigNullFieldsFix {
 
 				if (targetValue == null && sourceValue != null) {
 					field.set(target, sourceValue);
-				} else if (isFixable(field.getType())) {
-					if (targetValue != null && sourceValue != null) {
-						fixNullFields(targetValue, sourceValue);
-					}
+				} else if (targetValue != null && sourceValue != null && isFixable(field.getType())) {
+					fixNullFields(targetValue, sourceValue);
 				}
 			}
 		}

--- a/src/main/java/de/hysky/skyblocker/config/ConfigNullFieldsFix.java
+++ b/src/main/java/de/hysky/skyblocker/config/ConfigNullFieldsFix.java
@@ -1,0 +1,65 @@
+package de.hysky.skyblocker.config;
+
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.Map;
+
+import org.slf4j.Logger;
+
+import com.mojang.logging.LogUtils;
+
+import dev.isxander.yacl3.config.v2.api.SerialEntry;
+
+/**
+ * While this sounds like a data fixer it isn't. - It's the only reasonable solution to deal with the mine field
+ * that is YACL's null handling.
+ */
+public class ConfigNullFieldsFix {
+	private static final Logger LOGGER = LogUtils.getLogger();
+	private static final String CONFIGS_PACKAGE = "de.hysky.skyblocker.config.configs";
+
+	public static void init() {
+		SkyblockerConfig current = SkyblockerConfigManager.get();
+		SkyblockerConfig clean = new SkyblockerConfig();
+
+		try {
+			fixNullFields(current, clean);
+			SkyblockerConfigManager.save();
+		} catch (Exception e) {
+			LOGGER.error("[Skyblocker Config Null Fields Fixer] Failed to ensure that the config has no null fields! You may encounter crashes :(", e);
+		}
+	}
+
+	/**
+	 * Traverse through every config field to ensure that is isn't null, if it is then reset the value.
+	 */
+	private static void fixNullFields(Object target, Object source) throws Exception {
+		for (Field field : target.getClass().getDeclaredFields()) {
+			if (field.isAnnotationPresent(SerialEntry.class)) {
+				field.setAccessible(true);
+
+				Object targetValue = field.get(target);
+				Object sourceValue = field.get(source);
+
+				if (targetValue == null && sourceValue != null) {
+					field.set(target, sourceValue);
+				} else if (isFixable(field.getType())) {
+					if (targetValue != null && sourceValue != null) {
+						fixNullFields(targetValue, sourceValue);
+					}
+				}
+			}
+		}
+	}
+
+	private static boolean isFixable(Class<?> clazz) {
+		return !clazz.isPrimitive()
+				&& !clazz.isEnum()
+				&& !clazz.isRecord()
+				&& !clazz.equals(String.class)
+				&& !Number.class.isAssignableFrom(clazz)
+				&& !Map.class.isAssignableFrom(clazz)
+				&& !Collection.class.isAssignableFrom(clazz)
+				&& clazz.getPackageName().startsWith(CONFIGS_PACKAGE);
+	}
+}

